### PR TITLE
[5] add parallel remove command

### PR DESCRIPTION
**Pull request by jasoncoposky**

### CODE REVIEW

**Comment by trel** [comment]
is this also part of #6?

**Comment by jasoncoposky** [comment]
no this is a separate PR.  The beams were crossed, and should now be fixed.

File: commands/rm/main.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+        auto help_text() const noexcept -> std::string_view override
+        {
+            auto help =R"(
+Given a fully qualfied or relative path, remove the collection or object
+
+irods rm [options] fully_qualified_logical_path
+
+      --unregister        : unregister data instead of unlinking data
+      --no_trash          : do no move items to the trash can
```
Comment (trel) [inline commands/rm/main.cpp]: need more t ... "do not move"

Comment (jasoncoposky) [reply commands/rm/main.cpp]: fixed

**Comment by trel** [comment]
cool, please #

File: src/main.cpp [hunk 2]
```diff
@@ -110,18 +110,23 @@ auto load_cli_command_plugins(const po::variables_map& vm) -> cli_command_map_ty
...
-            namespace dll = boost::dll;
-            auto cli_impl = dll::import<irods::cli::command>(e.path(), "cli_impl", dll::load_mode::append_decorations);
-            map.insert_or_assign(cli_impl->name(), cli_impl);
+            try {
+                namespace dll = boost::dll;
+                auto cli_impl = dll::import<irods::cli::command>(e.path(), "cli_impl", dll::load_mode::append_decorations);
+                map.insert_or_assign(cli_impl->name(), cli_impl);
+            }
+            catch(...) {
+            }
```
Comment (korydraughn) [inline src/main.cpp]: why is a try-catch block needed here?  if loading a plugin throws an exception, the try-catch block in main() catches the exception.

Comment (jasoncoposky) [reply src/main.cpp]: This was me trying to bootstrap the CLI when there were some issues getting started, I can remove it.

**Comment by trel** [comment]
i see a merge conflict with CMakeLists.txt

**Pull request merged**